packages/venia-concept/src/components/FilterModal/filterModal.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import FilterFooter from './FilterFooter';
+import PropTypes from 'prop-types';
+import { List } from '@magento/peregrine';
+import { FiltersCurrent } from './FiltersCurrent';
+import classify from 'src/classify';
+import CloseIcon from 'react-feather/dist/icons/x';
+import Icon from 'src/components/Icon';
+import FilterBlock from './filterBlock';
+import defaultClasses from './filterModal.css';
+
+class FilterModal extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            root: PropTypes.string,
+            modalWrapper: PropTypes.string,
+            header: PropTypes.string,
+            headerTitle: PropTypes.string,
+            filterOptionsContainer: PropTypes.string
+        }),
+        filters: PropTypes.arrayOf(
+            PropTypes.shape({
+                request_var: PropTypes.string,
+                items: PropTypes.array
+            })
+        ),
+        filterAdd: PropTypes.func,
+        filterRemove: PropTypes.func,
+        closeModalHandler: PropTypes.func
+    };
+
+    getFilterParams = location => {
+        const params = new URLSearchParams(location.search);
+        let titles,
+            values = [];
+
+        let urlFilterParams = {};
+
+        for (var key of params.keys()) {
+            const cleanKey = key.replace(/\[\]\[.*\]/gm, '');
+
+            if (urlFilterParams[cleanKey]) continue;
+
+            titles = params.getAll(`${cleanKey}[][title]`);
+            values = params.getAll(`${cleanKey}[][value]`);
+
+            urlFilterParams[cleanKey] = titles.map((title, index) => ({
+                title: title,
+                value: values[index]
+            }));
+        }
+
+        return urlFilterParams;
+    };
+
+    componentDidMount = () => {
+        const filterParams = this.getFilterParams(this.props.history.location);
+        for (var key in filterParams) {
+            if (filterParams.hasOwnProperty(key)) {
+                filterParams[key].map(({ title, value }) => {
+                    this.props.filterAdd({
+                        group: key,
+                        title,
+                        value
+                    });
+                });
+            }
+        }
+    };
+
+    render() {
+        const { classes, isModalOpen, closeModalHandler } = this.props;
+        const modalClass = isModalOpen ? classes.rootOpen : classes.root;
```

Comment:
We don't need to thread this through from the root of the app. Instead, you could connect this component to the redux store and check the drawer like:

```js
mapStateToProps = ({app}) => {
 return {
    isActive: app.drawer === 'filter',
    ..otherProps
  }
})
```

Although now that I look at it, maybe @jimbo has a reason why we should connect the `RootComponents` but not the children of those components... I need to hear from him >.<

Reply:
Fixed in latest commit